tests/Browser/BlogManageTest.php

```diff
@@ -0,0 +1,562 @@
+        });
+    }
+
+     /**
+     * Test screen Edit blog by user.
+     * case 1.3: Edit blog: Test screen Edit blog by user
+     *
+     * @return void
+     */
+    public function testScreenEditBlog()
+    {
+        $user = $this->userCanSee();
+        $data = $this->getDataTest();
+        $this->browse(function ($browser) use ($user, $data) {
+            $browser->visit($data['path']['login'])
+                    ->loginAs($user)
+                    ->visit($data['path']['blog'])
+                    ->click('#blog-edit-2')
+                    ->assertPathIs('/blog/2/edit');
+        });
+    }
+
+    /**
+     * Test screen delete blog by user.
+     * case 1.4: Delete blog: Test screen delete blog by user
+     *
+     * @return void
+     */
+    public function testScreenDeleteBlog()
+    {
+        $user = $this->userCanSeeAndDelete();
+        $data = $this->getDataTest();
+        $this->browse(function ($browser) use ($user, $data) {
+            $browser->visit($data['path']['login'])
+                    ->loginAs($user)
+                    ->visit($data['path']['blog'])
+                    ->click('#blog-delete-5')
+                    ->waitFor('#deleteModal5')
+                    ->assertSee('Delete this blog?');
+        });
+    }
+
+    /**
+     * Blog list have not permission see blog
+     * case 1.5: Blog list have not permission see blog
+     *
+     * @return void
+     */
+    public function testBlogListUseHaveNotPermissionSeeBlog()
+    {
+        $user = $this->userCannotSee();
+        $data = $this->getDataTest();
+        $this->browse(function ($browser) use ($user, $data) {
+            $browser->visit($data['path']['login'])
+                    ->loginAs($user)
+                    ->visit($data['path']['blog'])
+                    ->assertPathIsNot($data['path']['blog']);
+        });
+    }
+
+    /**
+     * Blog list: User have not permission delete blog
+     * case 1.6: Blog list: User have not permission delete blog
+     *
+     * @return void
+     */
+    public function testBlogListUseHaveNotPermissionDeleteBlog()
+    {
+        $user = $this->userCanSee();
+        $data = $this->getDataTest();
+        $this->browse(function ($browser) use ($user, $data) {
+            $browser->visit($data['path']['login'])
+                    ->loginAs($user)
+                    ->visit($data['path']['blog'])
+                    ->assertDontSee('Delete');
+        });
+    }
+
+    /**
+     * Test Enter url not defined in route
+     * case 1.7: Enter url not defined in route
+     *
+     * @return void
+     */
+    public function testLoginUrlNotExist()
+    {
+        $user = $this->userCanSee();
+        $data = $this->getDataTest();
+        $this->browse(function ($browser) use ($user, $data) {
+            $browser->visit($data['path']['login'])
+                    ->loginAs($user)
+                    ->visit('/urlkhongtontai')
+                    ->assertSee('404');
```

Comment:
'404' is a text ? what if url have 404 content (ex : 'Nhung research lien quan den loi 404') ? please check for specific item that for 404

Reply:
I fixed